tests/test_cms_menus.py

```diff
@@ -250,6 +253,115 @@ def test_get_roots_with_versioning_disabled(self):
             list(roots), [menucontent_1.root, menucontent_2.root, menucontent_3.root]
         )
 
+    def test_draft_menu_on_draft_page(self):
+        """
+        Draft menu changes on draft page rendering.
+        """
+        menu_cont_draft = factories.MenuContentWithVersionFactory(version__state=DRAFT, language=self.language)
+        menu_cont_published = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        pagecontent_aaa = factories.PageContentWithVersionFactory(
```

Comment:
This test will only work if the pages share the same grouper (page). The published version has to be added first too as this is logically how the versions are created.